### PR TITLE
feat: create a error system to notify the user when something goes wr…

### LINKED ISSUE #16: feat: notify when the slices don't follow the sync function rules
## Description
There are some rules to be followed  to use the synchronization action and create new variants:
- use an acceptable type element for each slice (e.g. border-radius only accepts rectangle element)
- use no-mixed value (e.g. rectangle with 4 different border-radius)
- use a unique name for each slice value ( e.g. the user can't create two values with the "XL" name)

If these rules will be not followed, the plugin can't create the variants of those. Creating an error system that gives the user specific feedback using the Figma `notify` would be great. In this way, the user can fix these problems.

image

### PR DESCRIPTION
**Pull request by lucacacciarru**
# Proposed changes
create an error system to notify the user when something goes wrong in the syncTheme function

closes #16 

## Types of changes

- [X] ✨ New feature (non-breaking change which adds functionality or enhancements)
- [ ] 🐛 Bugfix (non-breaking change which fixes an issue)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ✅ Tests (added tests for an existing feature)
- [ ] 📚 Chore / Documentation Update (if none of the other choices apply)
- [ ] 🙌 Other (please, write a clear and concise description of the proposal in the section above)

## Checklist

- [X] :scissors: I have ensured that the PR is concise and broken down if needed
- [X] 👀 I have read the [README](../README.md) doc
- [X] 🧪 I have added tests that prove my fix is effective or that my feature works
- [ ] 📚 I have added necessary documentation (if appropriate)
- [X] 🔀 Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/plugin/syncTheme/utils.ts [hunk 1]
```diff
@@ -142,3 +180,64 @@ export const checkAndRemoveDeletedSlices = ({
...
+
+export type ErrorMap = Record<ErrorType, string[]>;
+
+/**
+ * Notify the user of each error found stored in the errorMap
+ *
+ * @param errorMap - Object containing the error types and their messages
+ */
+export const notifyErrorMessages = (errorMap: ErrorMap) => {
+  console.log(Object.values(errorMap));
```
Comment (gabrieleAngius) [inline src/plugin/syncTheme/utils.ts]: **todo:** remove this console.log

File: src/plugin/syncTheme/utils.ts [hunk 2]
```diff
@@ -142,3 +180,64 @@ export const checkAndRemoveDeletedSlices = ({
...
+function createWrongElementMessage({
+  acceptedElementType,
+  currentType,
+  sliceName,
+}: {
+  acceptedElementType: string;
+  currentType: string;
+  sliceName: string;
+}) {
+  return `The ${sliceName} slice accepts only ${acceptedElementType} elements, you're also using ${currentType}`;
```
Comment (gabrieleAngius) [inline src/plugin/syncTheme/utils.ts]: **suggestion:** What about `The ${sliceName} slice accepts only ${acceptedElementType} elements. Please remove ${currentType} elements` ?

File: src/plugin/syncTheme/utils.ts [hunk 3]
```diff
@@ -142,3 +180,64 @@ export const checkAndRemoveDeletedSlices = ({
...
+export function generateAndRecordErrorMessage<T extends ErrorType>({
+  errorType,
+  errorMap,
+  values,
+}: {
+  errorType: T;
+  errorMap: ErrorMap;
+  values: Parameters<typeof warningMessageGenerators[T]>[0];
+}) {
+  const selectedMessage = warningMessageGenerators[errorType](values as any);
```
Comment (gabrieleAngius) [inline src/plugin/syncTheme/utils.ts:240]: **question:** any chance to have a better type on `values`? I would avoid `as any` if possible

**Review by gabrieleAngius — CHANGES_REQUESTED** [review]

File: src/plugin/syncTheme/utils.ts (hunk 2 above)
Comment (lucacacciarru) [reply src/plugin/syncTheme/utils.ts]: Sounds good

**Review by gabrieleAngius — APPROVED** [review]

**Pull request merged**